dialer/utils.go

```diff
@@ -0,0 +1,80 @@
+package dialer
+
+import (
+	"fmt"
+
+	"github.com/gocql/gocql/internal/murmur"
+)
+
+type frameOp byte
+
+const (
+	// header ops
+	opError         frameOp = 0x00
+	opStartup       frameOp = 0x01
+	opReady         frameOp = 0x02
+	opAuthenticate  frameOp = 0x03
+	opOptions       frameOp = 0x05
+	opSupported     frameOp = 0x06
+	opQuery         frameOp = 0x07
+	opResult        frameOp = 0x08
+	opPrepare       frameOp = 0x09
+	opExecute       frameOp = 0x0A
+	opRegister      frameOp = 0x0B
+	opEvent         frameOp = 0x0C
+	opBatch         frameOp = 0x0D
+	opAuthChallenge frameOp = 0x0E
+	opAuthResponse  frameOp = 0x0F
+	opAuthSuccess   frameOp = 0x10
+)
+
+func GetFrameHash(frame []byte) int64 {
+	var p int
+	if frame[0] > 0x02 {
+		p = 1
+		frame[2] = byte('0')
+		frame[3] = byte('0')
+	} else {
+		p = 0
+		frame[2] = byte('0')
+	}
+	switch frame[3+p] {
+	case byte(opStartup):
+		return murmur.Murmur3H1(frame[:8+p])
+	case byte(opPrepare):
+		return murmur.Murmur3H1(frame)
+	case byte(opAuthResponse):
+		return murmur.Murmur3H1(frame)
+	case byte(opQuery):
+		//omit header and custom payload
+		index := 8 + p
+		customPayloadLenght := int(frame[8+p])<<8 | int(frame[9+p])
+		if customPayloadLenght > 0 {
+			index = index + 2
+		}
+		for i := 0; i < customPayloadLenght; i++ {
+			stringLenght := int(frame[index])<<8 | int(frame[index+1])
+			index = index + 2 + stringLenght
+			bytesLength := int(frame[index+0])<<24 | int(frame[index+1])<<16 | int(frame[index+2])<<8 | int(frame[index+3])
+			index = index + 4 + bytesLength
+		}
+		fmt.Println("INDEX: ", index)
+		statementLength := int(frame[index+0])<<24 | int(frame[index+1])<<16 | int(frame[index+2])<<8 | int(frame[index+3])
+		fmt.Println("STMT LEN: ", statementLength)
+		fmt.Println("FRAME LEN: ", len(frame))
+		return murmur.Murmur3H1(frame[index : statementLength+4])
```

Comment:
You need to exclude timestamp, if it is there (`if flags & flagDefaultTimestamp > 0` ). Take a look at `framer.writeQueryParams` for more details: https://github.com/scylladb/gocql/blob/57a9143688f0e4932333a32a4b61534a974a6364/frame.go#L1570-L1583

Reply:
for opQuery I only take statement in consideration, I omit whole QueryParams part so there is no need to exclude timestamp, should I take queryParams into the hash?

Reply:
Yes, I believe you should